src/components/TvDetails/index.tsx

```diff
@@ -451,12 +459,21 @@ const TvDetails: React.FC<TvDetailsProps> = ({ tv }) => {
                     />
                   </svg>
                   <span>
-                    {data.mediaInfo?.plexUrl
-                      ? intl.formatMessage(messages.playonplex)
-                      : data.mediaInfo?.plexUrl4k &&
-                        (hasPermission(Permission.REQUEST_4K) ||
-                          hasPermission(Permission.REQUEST_4K_TV))
-                      ? intl.formatMessage(messages.play4konplex)
+                    {data.mediaInfo?.plexUrl || data.mediaInfo?.jellyfinUrl
+                      ? intl.formatMessage(
+                          data.mediaInfo.plexUrl != undefined
+                            ? messages.playonplex
+                            : messages.playonjellyfin
+                        )
+                      : data.mediaInfo?.plexUrl4k ||
+                        (data.mediaInfo?.jellyfinUrl4k &&
+                          (hasPermission(Permission.REQUEST_4K) ||
+                            hasPermission(Permission.REQUEST_4K_MOVIE)))
+                      ? intl.formatMessage(
+                          data.mediaInfo.plexUrl != undefined
+                            ? messages.playonplex
+                            : messages.playonjellyfin
+                        )
```

Comment:
Maybe we should even extract the logic? This is barely readable for me and I wrote the plexUrl lol. What do you think?

Reply:
I think we should try to minimize the changes here, since it's likely we'll introduce some bugs.  Simply changing references to `plexUrl` to `mediaUrl` should be sufficient if we add `setMediaUrl()` to `Media`.